src/utils.js

```diff
@@ -466,21 +466,32 @@ function withDependentSchema(
   dependencyKey,
   dependencyValue
 ) {
-  let { oneOf, ...dependentSchema } = retrieveSchema(
+  let { oneOf, allOf, ...dependentSchema } = retrieveSchema(
```

Comment:
When both `oneOf` and `allOf` are present:

* The if/else below will only apply to `oneOf`.
* `allOf` is omitted from `dependentSchema`.

Is this intentional?

Reply:
Hmmm, that's true about the if/else thing. I'm not really sure what to do there. I'm not sure I _want_ to support something like:

```json
"dependencies": {
  "oneOf": [{ ... }, { ... }],
  "allOf": [{ ... }, { ... }]
}
```

But maybe that's naive. I definitely think for `allOf`, `oneOf`, and `anyOf`, there's definitely room to add support for combining all of them. However, I think this patch serves _our_ needs for now.

As for omitting `allOf` from the `dependentSchema`, that's intentional. It's supposed to pull it out, find the valid subschemas, and merge them into the main schema.

The remaining schemas in `dependentSchema` are all basic schemas that just automatically get merged into the schema on line 474.